src/App.tsx

```diff
@@ -1,54 +1,97 @@
+      clearTimeout(timer);
+      timer = window.setTimeout(f, delay, ...args);
+    };
+  }
+
+  const applyQuery = useCallback(
+    debounce(
+      setAppliedQuery,
+    ),
+    [],
+  );
+
+  const getVisiblePeople = () => {
+    return people.filter((person) => person
+      .name
+      .toLowerCase()
+      .includes(appliedQuery.toLowerCase()));
+  };
+
+  const visiblePeople = useMemo(
+    getVisiblePeople,
+    [people, appliedQuery],
+  );
+
+  const onSelected = (
+    personName: string,
+    dateBorn: number,
+    dateDied: number,
+  ) => {
+    setPersonPicked(personName);
+    setHeader(`${personName} (${dateBorn} - ${dateDied})`);
+    setPersonPicked('');
+    setDropdownActive(false);
+  };
+
+  const onChange = (event:React.ChangeEvent<HTMLInputElement>) => {
+    setPersonPicked(event.target.value);
+    applyQuery(event.target.value);
+  };
 
   return (
     <main className="section">
       <h1 className="title">
-        {`${name} (${born} = ${died})`}
+        {header}
       </h1>
 
-      <div className="dropdown is-active">
+      <div className={cn('dropdown', { 'is-active': dropdownActive })}>
         <div className="dropdown-trigger">
           <input
             type="text"
             placeholder="Enter a part of the name"
             className="input"
+            value={personPicked}
+            onFocus={() => setDropdownActive(!dropdownActive)}
+            onChange={onChange}
+            onBlur={() => setDropdownActive(false)}
           />
         </div>
 
         <div className="dropdown-menu" role="menu">
           <div className="dropdown-content">
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Bernard Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Antone Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter de Decker</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Petronella de Decker</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Hercke</p>
-            </div>
+            {visiblePeople.map((person) => (
+              <div
+                key={person.name}
+                className={cn('dropdown-item',
+                  { 'has-background-light': hoverActive === person.name })}
+                onMouseEnter={() => setHoverActive(person.name)}
+                onMouseDown={() => onSelected(person.name,
+                  person.born,
+                  person.died)}
```

Comment:
```suggestion
                onMouseDown={() => {
                  onSelected(
                    person.name,
                    person.born,
                    person.died
                  )}
                }
```

Reply:
<img width="1091" alt="Screen Shot 2024-01-15 at 12 13 57 PM" src="https://github.com/mate-academy/react_autocomplete/assets/119886901/c9718781-bc75-4e81-8a29-34e4112f2e4f">

not sure why it doesn't show as fixed